src/python/pants/engine/fs.py

```diff
@@ -1,21 +1,46 @@
-# Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-import os
 from dataclasses import dataclass
+from enum import Enum
 from typing import TYPE_CHECKING, Iterable, Optional, Tuple
 
 from pants.engine.collection import Collection
 from pants.engine.rules import RootRule, side_effecting
-from pants.option.custom_types import GlobExpansionConjunction as GlobExpansionConjunction
-from pants.option.global_options import GlobMatchErrorBehavior as GlobMatchErrorBehavior
-from pants.util.dirutil import maybe_read_file, safe_delete, safe_file_dump
 from pants.util.meta import frozen_after_init
 
 if TYPE_CHECKING:
     from pants.engine.internals.scheduler import SchedulerSession
 
 
+@dataclass(frozen=True)
+class Digest:
+    """A Digest is a content-digest fingerprint, and a length of underlying content.
+
+    Typically, this is used as a lightweight reference to a set of files known about by the engine.
```

Comment:
Rather than "typically", probably "always". If that wasn't true it would be a dangerously un-typesafe API. It should only ever not be true on the rust side.

Reply:
Sounds good. Will fix in a followup. Thanks!